### PR TITLE
Update Fallback Avatar to Support Themes 

### PR DESCRIPTION
**Pull request by grgia**
### Details

This PR updates the fallback avatar to use the same coloring pattern we use for coloring the SVGs for the default workspace avatars 

Example of the themes:

https://github.com/Expensify/App/assets/38015950/d8ed51a6-78f1-4925-b9d7-b5989e36e211

Also changes the loading color from offline to borders. See https://github.com/Expensify/App/pull/38674#issuecomment-2024192238

image

<summary>### NOTE</summary>

This PR only handles the SEARCH Bar, it does not fix this behavior all over the app (ie. chatting new users, manual request options, etc). That will be completed as part of this PR:
https://github.com/Expensify/App/issues/38743

Outdated: 

Example of users I know and don't know:

https://github.com/Expensify/App/assets/38015950/51ab8934-e538-4c1f-8fe2-ec327425b500

If you don't know a user, don't have an account in ONYX, or in any cases where you'd only have an optimisticID, Then we will default to the fallback avatar. 

</details>

<details>

### Fixed Issues

$ https://github.com/Expensify/App/issues/33161
https://github.com/Expensify/App/issues/34000
https://github.com/Expensify/App/issues/33470 (may impact testing steps for this one)

### Tests

- [X] Verify that no errors appear in the JS console
- [X] Same as QA

### QA Steps

- [X] Verify that no errors appear in the JS console
- [X] Trigger the fallback avatar to load by triggering an image error by going offline before loading the avatar of a user A with a cutom-uploaded avatar
- [X] ✅ Ensure that the fallback avatar is shown for that user 
- [X] Toggle to the opposite theme 
- [X] ✅ Ensure that the fallback avatar changes colors to match the theme (see description for video if you need a reference)

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/libs/OptionsListUtils.ts [hunk 1]
```diff
@@ -1682,10 +1683,10 @@ function getOptions(
         // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
         userToInvite.alternateText = userToInvite.alternateText || searchValue;
 
-        // If user doesn't exist, use a default avatar
+        // If user doesn't exist, always use the fallback avatar
         userToInvite.icons = [
             {
-                source: UserUtils.getAvatar('', optimisticAccountID),
+                source: FallbackAvatar,
```
Comment (s77rt) [inline src/libs/OptionsListUtils.ts:1689]:
```suggestion
```
Not specifying any source should by design fallback to the fallback avatart

File: src/components/Avatar.tsx [hunk 2]
```diff
@@ -107,13 +114,8 @@ function Avatar({
...
                         height={iconSize}
                         width={iconSize}
-                        fill={imageError ? theme.offline : iconFillColor}
-                        additionalStyles={[
-                            StyleUtils.getAvatarBorderStyle(size, type),
-                            isWorkspace && StyleUtils.getDefaultWorkspaceAvatarColor(name),
-                            imageError && StyleUtils.getBackgroundColorStyle(theme.fallbackIconColor),
-                            iconAdditionalStyles,
-                        ]}
+                        fill={imageError ? iconColors?.fill ?? theme.offline : fill}
```
Comment (s77rt) [inline src/components/Avatar.tsx:117]:
```suggestion
                        fill={imageError ? theme.offline : iconColors?.fill ?? fill}
```

On error we should use always the offline color

File: src/libs/OptionsListUtils.ts (hunk 1 above)
Comment (s77rt) [reply src/libs/OptionsListUtils.ts:1689]: I also found setting the source here causes imageError in Avatar in being true

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by dubielzyk-expensify** [comment]
Great stuff! There's this very brief flicker that happens between some of the searches. Could we eliminate this? 

https://github.com/Expensify/App/assets/145416827/654965b0-4290-4626-b948-6ca304b4bb62

Frozen down I see this:
CleanShot 2024-03-25 at 09 26 29@2x

Looks like it's the `icon` background color. If that's the default, then let's put it to `button` or `borders` so it matches the default avatar icon. I also see this icon color when I scroll my search very quickly before the avatars have loaded.

File: src/components/Avatar.tsx [hunk 3]
```diff
@@ -74,21 +74,28 @@ function Avatar({
...
-
     const isWorkspace = type === CONST.ICON_TYPE_WORKSPACE;
     const iconSize = StyleUtils.getAvatarSize(size);
 
     const imageStyle: StyleProp<ImageStyle> = [StyleUtils.getAvatarStyle(size), imageStyles, styles.noBorderRadius];
     const iconStyle = imageStyles ? [StyleUtils.getAvatarStyle(size), styles.bgTransparent, imageStyles] : undefined;
 
-    const iconFillColor = isWorkspace ? StyleUtils.getDefaultWorkspaceAvatarColor(name).fill : fill;
+    // We pass the color styles down to the SVG for the workspace and fallback avatar.
+    const useFallBackAvatar = imageError || source === Expensicons.FallbackAvatar;
```
[the remark this answers is not in the document]
Comment (grgia) [reply src/components/Avatar.tsx:84]: I agree with this idea, but I'm worried there's a few places in app that are setting the fallback avatar as the avatar. For example [here](https://github.com/Expensify/App/blob/georgia-fallbackAvatars/src/libs/UserUtils.ts#L86-L88). Maybe we should just return nothing?

[the remark this answers is not in the document]
Comment (grgia) [reply src/components/Avatar.tsx:84]: https://github.com/Expensify/App/blob/georgia-fallbackAvatars/src/libs/ReportUtils.ts#L1606-L1614

[the remark this answers is not in the document]
Comment (grgia) [reply src/components/Avatar.tsx:84]: Seeing as this gets a little hazy, what if I remove this change from the PR (leaving only the theme support) and we handle all the fallback issues in https://github.com/Expensify/App/issues/38743

[the remark this answers is not in the document]
Comment (s77rt) [reply src/components/Avatar.tsx:84]: Handling this separately sounds good to me 👍

**Comment by grgia** [comment]
@s77rt okay, I removed the search changes from this PR and pushed suggestions. I've started a new test build :)

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome dark
mweb-chrome light

</details>

<details>
<summary>iOS: Native</summary>


ios dark
ios light

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari dark
mweb-safari light

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web dark
web light

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop dark
desktop light

</details>

**Comment by s77rt** [comment]
Added screenshots except for Android. Still working on it...

**Comment by s77rt** [comment]
@grgia Can you please update the test steps?

**Review by s77rt — APPROVED** [review]

**Comment by dubielzyk-expensify** [comment]
Is the issue in [this comment](https://github.com/Expensify/App/pull/38674#issuecomment-2016988938) fixed cause I still see it in the test build!?

**Comment by grgia** [comment]
> Is the issue in https://github.com/Expensify/App/pull/38674#issuecomment-2016988938 fixed cause I still see it in the test build!?

@dubielzyk-expensify Should we just use a transparent fill instead? We removed the changes to the search fallback behavior from this PR to handle in an external PR

**Comment by grgia** [comment]
Here's an example of the loading custom avatar- @dubielzyk-expensify 

https://github.com/Expensify/App/assets/38015950/220eccf0-e0c0-4ea7-b71c-64cd33f04388

**Comment by dubielzyk-expensify** [comment]
I'm not sure what that means 😅  The key thing is that it looks like this:
CleanShot 2024-03-28 at 10 23 19@2x

cc @Expensify/design in case they feel strongly against

File: src/components/Avatar.tsx [hunk 4]
```diff
@@ -112,7 +112,7 @@ function Avatar({
                         src={avatarSource}
                         height={iconSize}
                         width={iconSize}
-                        fill={imageError ? theme.offline : iconColors?.fill ?? fill}
+                        fill={imageError ? theme.border : iconColors?.fill ?? fill}
```
Comment (s77rt) [inline src/components/Avatar.tsx:115]: Revert. Instead update the background color returned by `getAvatarStyle`
https://github.com/Expensify/App/blob/469b11237c3079232d9bbf65899c3a4caa387b26/src/styles/utils/index.ts#L1228-L1236

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/components/Avatar.tsx (hunk 4 above)
Comment (grgia) [reply src/components/Avatar.tsx:115]: https://github.com/Expensify/App/pull/38674#issuecomment-2024192238

Comment (grgia) [reply src/components/Avatar.tsx:115]: It should never be the offline color

Comment (s77rt) [reply src/components/Avatar.tsx:115]: That's the current code, not the suggestion.

Comment (grgia) [reply src/components/Avatar.tsx:115]: I changed this function too

Comment (s77rt) [reply src/components/Avatar.tsx:115]: Changing the color fill color here does not fix the outlined issue but causes regressions

Comment (s77rt) [reply src/components/Avatar.tsx:115]: Please revert. This causes a regression when offline

Screenshot 2024-03-28 at 3 12 37 PM


The change in getAvatarStyle is enough

https://github.com/Expensify/App/assets/16493223/084b0a54-252d-4f02-8689-7771f0a164b9

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
@grgia The above comment is not resolved, please check https://github.com/Expensify/App/pull/38674#discussion_r1543058090

File: src/components/Avatar.tsx (hunk 4 above)
Comment (grgia) [reply src/components/Avatar.tsx:115]: > Changing the color fill color here does not fix the outlined issue but causes regressions

hmm I don't think this line caused the regression. That's the correct fill color. But the icon should be the fallback icon right? I'll look at this today

**Comment by dubielzyk-expensify** [comment]
I'm trying to test and do a final review, but seeing this: 
CleanShot 2024-04-03 at 11 08 41@2x

**Comment by dubielzyk-expensify** [comment]
Woops. My bad

File: src/components/Avatar.tsx (hunk 4 above)
Comment (grgia) [reply src/components/Avatar.tsx:115]: Sorry @s77rt could you rephrase? I'm a bit confused. The changes I made reflect this design comment https://github.com/Expensify/App/pull/38674#issuecomment-2024192238

**Comment by grgia** [comment]
@dubielzyk-expensify I merged main and triggered a new build

**Review by dubielzyk-expensify — DISMISSED** [review]
Looks good to me from a design end 😄

**Comment by grgia** [comment]
@dubielzyk-expensify just confirming- should the offline color for the icon look like this? (when a user has an avatar that's not able to load)

image

File: src/components/Avatar.tsx (hunk 4 above)
Comment (s77rt) [reply src/components/Avatar.tsx:115]: There are 2 colors, the background color and the fill color. The fill color is used when we have an icon to render on top e.g. the person shape icon. Those two colors shouldn't be the same otherwise you won't see the icon (which the regression we have now).

> The changes I made reflect this design comment https://github.com/Expensify/App/pull/38674#issuecomment-2024192238

The requested change is to change the background color which is already done (changed background color in getAvatarStyle)

**Comment by dubielzyk-expensify** [comment]
When offline we should use our fallback avatar:
CleanShot 2024-04-04 at 13 05 21@2x

**Comment by madmax330** [comment]
@s77rt this is ready for final review

**Comment by s77rt** [comment]
@madmax330 Still discussing unresolved comment https://github.com/Expensify/App/pull/38674#discussion_r1550516966

File: src/components/Avatar.tsx (hunk 4 above)
Comment (grgia) [reply src/components/Avatar.tsx:115]: thank you for explaining, you're so right!

**Comment by grgia** [comment]
image
@s77rt fixed

**Comment by grgia** [comment]
Rebuilding for easy testing

**Review by s77rt — APPROVED** [review]

**Review by lakchote — APPROVED** [review]

**Review by madmax330 — APPROVED** [review]
